src/main/java/essenmakanan/parser/RecipeParser.java

```diff
@@ -218,16 +218,23 @@ public static RecipeIngredientList parseDataRecipeIngredients(String ingredients
         String[] parsedIngredients = ingredientsString.split(" , ");
         ArrayList<Ingredient> ingredientList = new ArrayList<>();
 
-        for (String ingredient : parsedIngredients) {
-            String[] parsedIngredient = ingredient.split(" \\| ");
+        for (String ingredientData : parsedIngredients) {
+            String[] parsedIngredient = ingredientData.split(" \\| ");
 
             if (parsedIngredient.length != 3 || parsedIngredient[1].isBlank()) {
                 throw new EssenStorageFormatException();
             }
 
-            String ingredientName = parsedIngredient[0];
-            Double ingredientQuantity = Double.parseDouble(parsedIngredient[1]);
-            IngredientUnit ingredientUnit = IngredientUnit.valueOf(parsedIngredient[2]);
+            String ingredientName = parsedIngredient[0].trim();
+
+            for (Ingredient ingredient : ingredientList) {
+                if (ingredient.getName().equals(ingredientName)) {
+                    throw new EssenStorageFormatException();
+                }
+            }
+
+            Double ingredientQuantity = Double.parseDouble(parsedIngredient[1].trim());
```

Comment:
would it be safer to check if ingredientQuantity is parsable? Perhaps in a try catch `NumberFormatException`